### ISSUE TITLE
Make code fully platform-agnostic

### ISSUE THREAD

**Issue by Technochips** [studio-minus/madness-interactive-reloaded#380]
**Description**
The code currently has a few references to a `WINDOWS` (and `LINUX`) preprocessor define. Currently, it seems to exclusively be used to mark the build as a "Windows build" in the credits page. My proposal is to remove these.

**Reasoning**
dotnet applications contains at least two things: the runtime, and the application's actual code.

The runtime is platform-dependant. A Windows build will ship with a Windows runtime. A Linux build will ship with a Linux runtime. The build can also ship with no runtime, in which case a runtime must be installed on the user's computer to run the application. The runtime itself is the same across every dotnet applications.

The application code (the one you get after compiling the game) is platform-independant and architecture-independant. It's not native assembly code, it uses code called "CLI", which gets interpreted and executed by the runtime. This is to allow the same binary to run correctly on any platforms. Checking if we're on a specific platform is done at runtime with `IsOSPlatform()`, which seems to be the preferred method of doing this on dotnet.

Currently, the use of preprocessor defines (`WINDOWS` and `LINUX`) breaks this, and ends up producing two different code builds that just so happen to have slightly different logic in three functions that turns it platform-dependant. The idea is to not do that.

**Additional context**
Currently, the AUR build runs something like `dotnet build -r linux-x64`, which is equivalent to "please build this and also ship in the linux runtime". The game works very well, but there's a single problem where it says that it's the Windows build on the credits screen, because I did not bother changing the defines. (I then learned that pubxml files are a thing, and I think I was supposed to use those).

**Comment by mestiez** [comment]
Yes this is a remnant of how things worked very early in development. I'll remove it.
